upgrades/upgrade_testing_framework/steps/step_start_source_cluster.py

```diff
@@ -0,0 +1,29 @@
+from upgrade_testing_framework.core.framework_step import FrameworkStep
+from upgrade_testing_framework.cluster_management.cluster import Cluster, ClusterNotStartedInTimeException
+
+class StartSourceCluster(FrameworkStep):
+    """
+    This step starts the source cluster and ensures it is healthy
+    """
+
+    def _run(self):
+        # Get the state we need
+        source_docker_image = self._get_state_value("test_config")["source_docker_image"]
+        docker_client = self.state.docker_client
+
+        # Begin the step body
+        self.logger.info("Creating source cluster...")
+        cluster = Cluster("source-cluster", 3, docker_client, source_docker_image)
+        cluster.start()
+
+        try: 
+            max_wait_sec = 30
+            self.logger.info(f"Waiting up to {max_wait_sec} sec for cluster to be active...")
+            cluster.wait_for_cluster_to_start_up(wait_limit_sec=30)
+        except ClusterNotStartedInTimeException as exception:
+            self.fail("Cluster did not start within time limit", exception)
```

Comment:
should anything be saved to the state in this case? we possibly have a cluster that was created but not started--should there be a record of it in the state?

Reply:
Good callout.  I feel like we need to have something like "cleanup" steps that are run in the equivalent of a `finally` block in the FrameworkRunner's main loop to ensure cleanup happens.  I'll think on that.

In this particular case, there isn't any value in saving the cluster to the state because the FrameworkState is a mess and most stuff isn't actually serialized to the `state-file`, so... maybe not?  I can see an argument to add that now so we don't forget to handle the edge case later though, even if there's no payoff now.